go/libkb/keylock.go

```diff
@@ -5,46 +5,62 @@ package libkb
 
 import "fmt"
 
-const WhichPassphraseKeybase = "your Keybase"
+type PassphraseType string
+
+const (
+	PassphraseTypeKeybase PassphraseType = "Keybase"
+	PassphraseTypePGP     PassphraseType = "PGP"
+)
+
+type UnlockerFunc func(pw string, storeSecret bool) (ret GenericKey, err error)
 
 type KeyUnlocker struct {
-	Tries          int
-	Reason         string
-	KeyDesc        string
-	Which          string
-	UseSecretStore bool
-	UI             SecretUI
-	Unlocker       func(pw string, storeSecret bool) (ret GenericKey, err error)
 	Contextified
+	tries          int
+	reason         string
+	keyDesc        string
+	which          PassphraseType
+	useSecretStore bool
+	ui             SecretUI
+	unlocker       UnlockerFunc
+}
+
+func NewKeyUnlocker(g *GlobalContext, tries int, reason string, keyDesc string, which PassphraseType, useSecretStore bool, ui SecretUI, unlocker UnlockerFunc) KeyUnlocker {
+	return KeyUnlocker{
+		Contextified:   NewContextified(g),
+		tries:          tries,
+		reason:         reason,
+		keyDesc:        keyDesc,
+		which:          which,
+		useSecretStore: useSecretStore,
+		ui:             ui,
+		unlocker:       unlocker,
+	}
 }
 
 func (arg KeyUnlocker) Run() (ret GenericKey, err error) {
 	var emsg string
 
-	which := arg.Which
-	if len(which) == 0 {
-		which = "the"
-	}
-	prompt := "Please enter " + which + " passphrase to unlock the secret key for:\n" +
-		arg.KeyDesc + "\n"
-	if len(arg.Reason) > 0 {
-		prompt = prompt + "\nReason: " + arg.Reason
+	if arg.ui == nil {
+		err = NoUIError{"secret"}
+		return nil, err
 	}
 
-	if arg.UI == nil {
-		err = NoUIError{"secret"}
-		return
+	prompt := "Please enter the " + string(arg.which) + " passphrase to unlock the secret key for:\n" +
```

Comment:
So this will say:

    Please enter *the* Keybase passphrase to unlock the secret key for:

and

    Please enter *the* PGP passphrase to unlock the secret key for:

Then title will be

    Your Keybase passphrase

or

    Your PGP passphrase

Perhaps `the` in prompt should be replaced with `your`?